pkg/network/cluster_config.go

```diff
@@ -117,9 +123,128 @@ func validateClusterConfig(clusterConfig *configv1.Network, infraRes *bootstrap.
 	}
 
 	if _, ok := clusterConfig.Annotations[names.NetworkTypeMigrationAnnotation]; ok {
-		// HostedControlPlane is not nil if in a HyperShift env
-		if infraRes.HostedControlPlane != nil {
-			return errors.Errorf("network type live migration is not supported on HyperShift clusters")
+		return validateLiveMigration(clusterConfig, infraRes, client)
+	}
+
+	return nil
+}
+
+// validateCIDROverlap validates whether any of the ClusterNetwork and ServiceNetwork CIDRs overlap with
+// the internal CIDRs used by OVNKubernetes
+func validateOVNKubernetesCIDROverlap(clusterCofnig *configv1.Network, operConfig *operv1.Network) error {
+	// Verify whether the available subnets conflict between CNIs
+	type subnet struct {
+		name string
+		cidr string
+	}
+	var subnets []subnet
+	for _, cn := range clusterCofnig.Spec.ClusterNetwork {
+		subnets = append(subnets, subnet{
+			name: "clusterNetwork",
+			cidr: cn.CIDR,
+		})
+	}
+	for _, svcNet := range clusterCofnig.Spec.ServiceNetwork {
+		subnets = append(subnets, subnet{
+			name: "serviceNetwork",
+			cidr: svcNet,
+		})
+	}
+
+	v4InternalSubnet, v6InternalSubnet := GetInternalSubnets(operConfig.Spec.DefaultNetwork.OVNKubernetesConfig)
+	subnets = append(subnets, subnet{
+		name: "v4InternalSubnet",
+		cidr: v4InternalSubnet,
+	})
+	subnets = append(subnets, subnet{
+		name: "v6InternalSubnet",
+		cidr: v6InternalSubnet,
+	})
+
+	v4InternalTransitSwitchSubnet, v6InternalTransitSwitchSubnet := GetTransitSwitchSubnets(operConfig.Spec.DefaultNetwork.OVNKubernetesConfig)
+	subnets = append(subnets, subnet{
+		name: "v4InternalTransitSwitchSubnet",
+		cidr: v4InternalTransitSwitchSubnet,
+	})
+	subnets = append(subnets, subnet{
+		name: "v6InternalTransitSwitchSubnet",
+		cidr: v6InternalTransitSwitchSubnet,
+	})
+
+	v4InternalMasqueradeSubnet, v6InternalMasqueradeSubnet := GetMasqueradeSubnet(operConfig.Spec.DefaultNetwork.OVNKubernetesConfig)
+	subnets = append(subnets, subnet{
+		name: "v4InternalMasqueradeSubnet",
+		cidr: v4InternalMasqueradeSubnet,
+	})
+	subnets = append(subnets, subnet{
+		name: "v6InternalMasqueradeSubnet",
+		cidr: v6InternalMasqueradeSubnet,
+	})
+
+	for i, subnetA := range subnets {
```

Comment:
shall we put the ovn-k internal subnets and the cluster subnets in separate lists and compare them with each other?

Reply:
I think we can keep it as is, it would detect overlaps between ovn-k networks if they were specified